### PR TITLE
Allow druid-kubernetes-overlord-extensions to be loaded in any druid service

### PR DESCRIPTION
**Pull request by nlippis**
#### Release note

<hr>

##### Key changed/added classes in this PR
 * `K8sOverlordModule.java`

<hr>

This PR has:

- [x] been self-reviewed.
- [ ] added documentation for new or modified features or behaviors.
- [ ] a release note entry in the PR description.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [x] been tested in a test Druid cluster.

### CODE REVIEW

**Review by clintropolis — APPROVED** [review]
:+1: this seems consistent with other modules that should alter behavior based on where they are run

I think you might get hit by the code coverage bot though. If that happens search for files ending with `ModuleTest` for examples which all typically do something with `Guice.createInjector` to make an injector and set some stuff and make sure the right things happen.

It looks like you also got caught up by the dependency bot for using `org.apache.commons.collections4` without explicitly adding it to the pom.

```
Warning:  Used undeclared dependencies found:
Warning:     org.apache.commons:commons-collections4:jar:4.2:provided
```
so either add to the pom or don't use it (it looks like `LagBasedAutoScaler` is the only thing using `org.apache.commons.collections4` in `druid-indexing-service`)

File: extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/K8sOverlordModule.java [hunk 1]
```diff
@@ -37,31 +41,42 @@
...
-        binder,
-        "druid.indexer.runner.type",
-        Key.get(TaskRunnerFactory.class),
-        Key.get(KubernetesTaskRunnerFactory.class)
-    );
-    final MapBinder<String, TaskRunnerFactory> biddy = PolyBind.optionBinder(
-        binder,
-        Key.get(TaskRunnerFactory.class)
-    );
+    if (SetUtils.emptyIfNull(nodeRoles).contains(NodeRole.OVERLORD)) {
```
Comment (paul-rogers) [inline extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/K8sOverlordModule.java:59]: It turns out that there is an easier way to do this:

```java
@LoadScope(roles = NodeRole.OVERLORD_JSON_NAME)
public class K8sOverlordModule implements DruidModule
```

Comment (nlippis) [reply extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/K8sOverlordModule.java:59]: Nice!

Comment (clintropolis) [reply extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/K8sOverlordModule.java]: neat :+1:

Comment (clintropolis) [reply extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/K8sOverlordModule.java]: oof, I even reviewed the change that added that, but totally forgot about it 🙃

Comment (nlippis) [reply extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/K8sOverlordModule.java]: Done

**Pull request merged**